### PR TITLE
Add Signature table functions

### PR DESCRIPTION
**Pull request by elimillera**
Most of this was taken from the scharp branch, just reformatting it for our workflow.

`vt_generate_sig_table` takes a data.frame of people and creates the data.frame for signitures. This should allow the user to access the signiture table if they need to.

`vt_render_sig_table` takes a data.frame of people, passes it though `vt_generate_sig_table` and puts it into a Rmarkdown file. This should work well if we need to put it into another markdown, or use the bookdown approach.

### CODE REVIEW

**Comment by thebioengineer** [comment]
Would we want to use the validation.yml to populate this?

File: R/sig_table.R [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#'
+#' @param people data frame with variables: role, name, and title
+#' @param template Name of the template to render
+#' @param output directory passed to `render_template()`
+#'
+#' @return The text to include in a validation R markdown document
+#'
+#' @export
+vt_render_sig_table <- function(people, template = "sig_table.Rmd", output) {
+  render_template(template = template,
```
Comment (thebioengineer) [inline R/sig_table.R:34]: should we add ability to add this into the validation.yml (#57 ) be default?

Comment (thebioengineer) [reply R/sig_table.R:34]: this is a more general thought to consider for all templates @elimillera @mariev

Comment (elimillera) [reply R/sig_table.R:34]: @thebioengineer Thats what I was thinking. I think the majority of the rendering could be specified in the config and the helper rendering functions would be used where they are specified

Comment (mariev) [reply R/sig_table.R:34]: If we write report rendering function correctly, we should just be able to cycle through each of the files in the config field `validation_files` in order.

My vote that the default report template includes the code to generate the various kable calls, but we should have the flexibility to accomodate a scenario where the parent report only contains the setup chunk and any rmd/md with signature tables, etc is in the first 1+ items of that `validation_files` field. 

One use case I can imagine is that the org has an approved form for capturing signatures which is required by SOP.

File: inst/templates/sig_table.Rmd [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Signitures
```
Comment (mariev) [inline inst/templates/sig_table.Rmd:1]: signature

Comment (elimillera) [reply inst/templates/sig_table.Rmd]: Updated, good catch!

**Review by mariev — CHANGES_REQUESTED** [review]
@elimillera I'm looking for a bit more to this branch before I can start incorporating it into building out a validation report.

1. How to get information from validation.yml to populate this?
2. Is there a way for users to build their own validation report within the knitr paradigm, without relying on a child rmd? Will they have to write a separate kable call with each report? Try incorporating lines 6+ from sig_table.Rmd into a function named `vt_kable_*`
3. test cases to handle when validation details are incomplete
4.  Consider a scenario where future users or developers will create variants of validation report templates, including writing different kable() calls. How do the scraping, reformatting, and rendering fit into this?
5.  What strategy are we using for rendering the validation report? @thebioengineer are we assuming that `vt_render_validation_report` uses the K-M vs M-K approach? edit: I believe the dynamic labeling locks us into the M-K paradigm.


Specific to the vt_render_sig_table function, what is the use case for creating extension-less output?

**Comment by elimillera** [comment]
> 1. How to get information from validation.yml to populate this?

That was something I wasn't sure about. Would we just want to pull in all of the names from the config file? All of the function authors as well? I'm not sure if this is something that is perfectly generalizable. I think a solid default would be to add it with all the usernames in the config file.

> 4. Consider a scenario where future users or developers will create variants of validation report templates, including writing different kable() calls. How do the scraping, reformatting, and rendering fit into this?

Would the `vt_generate_sig_table` not be a suitable drop in? I was thinking a developer could use that to create whatever format they needed. I would see the value in a function like `vt_kable_sig_table` that takes different arguments to format it, is that more consistent with how we are going to be rendering the report?

**Comment by mariev** [comment]
> Would we just want to pull in all of the names from the config file?
pull in all the names listed in the usernames field. 

>Would the vt_generate_sig_table not be a suitable drop in? 
No. calls to kable would be needed, and we don't have a nice default yet. The expectation is that users/future developers may want to write specific kable() calls for their reporting needs, but {valtools} will provide a bare bones functionality out of box.

>   I would see the value in a function like vt_kable_sig_table that takes different arguments to format it, is

At this time that extensibility is future work.  If there are args that we can quickly implement, go for it!

File: R/sig_table.R [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+#' @return A dataframe created from the validation config containing a row for
+#'   each user with the columns: role, name_and_title, signature, and date.
+#'
+#' @export
+#'
+#' @importFrom purrr map_dfr
+vt_generate_sig_table <- function(){
+
+  # Pull all the usernames into a data.frame
+  people <- map_dfr(read_validation_config()$usernames, data.frame)
```
Comment (thebioengineer) [inline R/sig_table.R:12]: dplyr/purrr is a rather large dependency to use this function. `do.call('rbind', lapply(read_validation_config()$usernames, data.frame))` achieves the same goal

Comment (elimillera) [reply R/sig_table.R:12]: Thanks @thebioengineer I was just thinking the same

**Comment by elimillera** [comment]
Hey @mariev,
I believe addressed most of your comments. Let me know if I misunderstood anything.
- It now pulls in the validation config for the users
- Now the `vt_generate_sig_table` gets the data.frame from the config file, `vt_kable_sig_table` holds the defaults for rendering kable, `vt_render_sig_table` creates the child rmd(and gives it the right extension).
- I've added a test for the `vt_kable_sig_table` function but let me know if there are any other tests you want me to add.

**Comment by mariev** [comment]
can you pull develop in again?

**Review by mariev — APPROVED** [review]
pushed some changes for c.i. after merging in the other `sig_table` branch.

**Review by thebioengineer — APPROVED** [review]

**Comment by mariev** [comment]
@elimillera @thebioengineer can this be merged? looks like all reviews are complete??

**Comment by thebioengineer** [comment]
Lets do it, and we can open any issues and make corrections in a future PR

**Pull request merged**